lib/components/popups/course-search/query-components/SearchList.tsx

```diff
@@ -59,42 +54,29 @@ const SearchList: FC<{
    */
   const courseList = () => {
     let toDisplay: any = [];
-    let startingIndex = pageNum * coursesPerPage;
-    let endingIndex =
-      startingIndex + coursesPerPage > filteredCourses.length
-        ? filteredCourses.length - 1
-        : startingIndex + coursesPerPage - 1;
-    for (let i = startingIndex; i <= endingIndex; i++) {
-      let inserted: string[] = [];
+    for (let i = 0; i < filteredCourses.length; i++) {
       const inspecting = { ...filteredCourses[i] };
-      inspecting.versions.forEach((v: any, versionNum: number) => {
-        let alreadyInserted = false;
-        inserted.forEach((term: string) => {
-          if (term.includes(v.term)) {
-            alreadyInserted = true;
-          }
-        });
+      for (let j = 0; j < inspecting.versions.length; j++) {
+        const v = inspecting.versions[j];
         if (
-          !alreadyInserted &&
-          (v.term === searchFilters.term + ' ' + searchFilters.year ||
-            (searchFilters.term === 'All' &&
-              (searchFilters.year === currentPlan.years[0].year ||
-                searchFilters.year.toString() === v.term.split(' ')[1])))
+          v.term === searchFilters.term + ' ' + searchFilters.year ||
+          (searchFilters.term === 'All' &&
+            (searchFilters.year === currentPlan.years[0].year ||
+              searchFilters.year.toString() === v.term.split(' ')[1]))
```

Comment:
find desired version (term + ' ' + year) 
one of them has to match because backend query guarantees so